inst/scripts/generateHGNCHash.R

```diff
@@ -0,0 +1,59 @@
+# generateHGNCHash.R
+
+#' Generates a data frame containing all valid HGNC gene symbols, calls fmatch
+#' on the data frame so it will create a hash of the data frame to be used by
+#' checkGeneSymbols.R.
+#'
+#' @param fURL The URL to the file containing valid HGNC gene symbols.
+#'
+#' @seealso \code{\link{isGeneSymbol}} is used to check if a gene symbol is a
+#' valid HGNC gene symbol that is contained in inst/extdata/HGNCSymbols.rds.
+
+library(fastmatch)
+
+# Open the connection/file.
+## Change fURL to be the URL of the file containing HGNC gene symbols.
+## The file is assumed to contain the symbol, status and locus group of the gene.
+fURL <- "http://tinyurl.com/kk3r9ck"
+HGNCFile <- url(fURL, open = "r")
+
+# Read the header and vertorize.
+header <- readLines(HGNCFile, n = 1)
+headerVector <- unlist(strsplit(header, "\t"))
+
+# Find the index of the columns that contain the gene symbol, status, and locus
+# group of the genes.
+symbolCol <- which(headerVector == "Approved Symbol")
+statusCol <- which(headerVector == "Status")
+locusCol <- which(headerVector == "Locus Group")
+
+# vectors containing the (important) statuses and locus groups of interest.
+impStatuses <- c("Approved")
+impLocusGroups <- c("protein-coding gene", "other", "phenotype")
```

Comment:
Well we never discussed which Locus Types to include. And in the email I said I think I should include locus groups "other" and "phenotype".

Here is the email, for reference:

> > Okay I think I get it. Also, here is what I got after calling unique on the different columns of the data obtained from HGNC.
> 
> > unique(status) 
> [1] "Approved"         "Symbol Withdrawn" "Entry Withdrawn" 
> > unique(locusType)
> [1] "gene with protein product"  "RNA, long non-coding"       "withdrawn"                  "pseudogene"          
> [5] "virus integration site"     "readthrough"                "phenotype only"             "unknown"                   
> [9] "region"                     "endogenous retrovirus"      "fragile site"               "immunoglobulin gene"       
> [13] "immunoglobulin pseudogene"  "transposable element"       "RNA, micro"                 "RNA, ribosomal"            
> [17] "RNA, transfer"              "complex locus constituent"  "protocadherin"              "RNA, cluster"              
> [21] "RNA, misc"                  "RNA, small nuclear"         "RNA, small cytoplasmic"     "RNA, small nucleolar"      
> [25] "RNA, Y"                     "T-cell receptor gene"       "T-cell receptor pseudogene" "RNA, vault"                
> > unique(locusGroup)
> [1] "protein-coding gene" "non-coding RNA"      "withdrawn"           "pseudogene"          "other"               
> [6] "phenotype"          
>
> Which genes would you say I should include in the data frame? I think the genes with Locus Group "other" and "phenotype" might also be important? What do you think?